src/state_space/error.rs

```diff
@@ -0,0 +1,20 @@
+use alloy::transports::TransportErrorKind;
+use thiserror::Error;
+
+use crate::amms::error::AMMError;
+
+#[derive(Error, Debug)]
+pub enum StateSpaceError {
+    #[error(transparent)]
+    AMMError(#[from] AMMError),
+    #[error(transparent)]
+    TransportError(#[from] alloy_json_rpc::RpcError<TransportErrorKind>),
+    #[error(transparent)]
+    JoinError(#[from] tokio::task::JoinError),
+    #[error(transparent)]
+    ErrReport(#[from] eyre::Report),
+    #[error("Logs Are Empty")]
+    MissingLogs,
+    #[error("Block Number Does not Exist")]
+    MissingBlockNumber,
```

Comment:
Are we able to remove this error and gracefully handle when block number is not found?

When is it possible to recieve a log without a block number? Does this mean that the log was not included in the block? If so, we should prefer `if let Some()`.


Here is the Log type for reference.
```rust
pub struct Log<T = LogData> {
    pub inner: Log<T>,
    pub block_hash: Option<FixedBytes<32>>,
    pub block_number: Option<u64>,
    pub block_timestamp: Option<u64>,
    pub transaction_hash: Option<FixedBytes<32>>,
    pub transaction_index: Option<u64>,
    pub log_index: Option<u64>,
    pub removed: bool,
}
```